app/backend/src/tests/test_admin.py

```diff
@@ -88,59 +88,96 @@ def test_AccessByNormalUser(db):
+
+        with _admin_session(_get_super_token()) as api:
+            res = api.GetUserDetails(admin_pb2.GetUserDetailsReq(user=normal_user.username))
+        assert res.user_id == normal_user.id
+        assert res.username == normal_user.username
+        assert res.email == normal_user.email
+        assert res.gender == normal_user.gender
+        assert res.banned == False
+        assert res.deleted == False
+
         with _admin_session(_get_super_token()) as api:
-            res = api.GetUserEmailById(admin_pb2.GetUserEmailByIdReq(user_id=normal_user.id))
-        assert res.email == NORMAL_USER_EMAIL
+            res = api.GetUserDetails(admin_pb2.GetUserDetailsReq(user=normal_user.email))
         assert res.user_id == normal_user.id
+        assert res.username == normal_user.username
+        assert res.email == normal_user.email
+        assert res.gender == normal_user.gender
+        assert res.banned == False
+        assert res.deleted == False
 
 
-def test_GetEmailByUserName(db):
+def test_ChangeUserGender(db):
     with session_scope() as session:
         _generate_normal_user(session)
         _generate_super_user(session)
         normal_user = _get_normal_user(session)
+
         with _admin_session(_get_super_token()) as api:
-            res = api.GetUserEmailByUsername(admin_pb2.GetUserEmailByUsernameReq(username=normal_user.username))
-            assert res.email == NORMAL_USER_EMAIL
-            assert res.user_id == normal_user.id
+            res = api.ChangeUserGender(admin_pb2.ChangeUserGenderReq(user=normal_user.username, gender="Machine"))
+        assert res.user_id == normal_user.id
+        assert res.username == normal_user.username
+        assert res.email == normal_user.email
+        assert res.gender == "Machine"
+        assert res.banned == False
+        assert res.deleted == False
 
 
-def test_GetBanUser(db):
+def test_BanUser(db):
     with session_scope() as session:
         _generate_normal_user(session)
         _generate_super_user(session)
+        normal_user = _get_normal_user(session)
+
         with _admin_session(_get_super_token()) as api:
-            normal_user = _get_normal_user(session)
-            api.BanUser(admin_pb2.BanUserReq(user_id=normal_user.id))
-            session.refresh(normal_user)
-            assert normal_user.is_banned
+            res = api.BanUser(admin_pb2.BanUserReq(user=normal_user.username))
+        assert res.user_id == normal_user.id
+        assert res.username == normal_user.username
+        assert res.email == normal_user.email
+        assert res.gender == normal_user.gender
+        assert res.banned == True
+        assert res.deleted == False
 
 
-def test_GetDeleteUser(db):
+def test_DeleteUser(db):
     with session_scope() as session:
         _generate_normal_user(session)
         _generate_super_user(session)
+        normal_user = _get_normal_user(session)
+
         with _admin_session(_get_super_token()) as api:
```

Comment:
Is there anyway for an outside user to abuse this by generating a super user and using the token to make admin requests?

Reply:
Nah these are just testing helpers, making someone a superuser requires changing it directly in the database